modulus/models/graphcast/graph_cast_net.py

```diff
@@ -114,16 +228,31 @@ class GraphCastNet(Module):
         passing no process group name leads to a parallelism across the default
         process group. Otherwise, the group size of a process group is expected
         to match partition_size.
-    expect_partitioned_input : bool, default=False,
+    use_lat_lon_partitioning : bool, default=False
```

Comment:
Not super important for GraphCast, since the graph structure is fixed here, but for MeshGraphNet, it might be good to figure out a way to expose the `GraphPartition` object or something similar so one can pass in a custom partition.

Reply:
MeshGraphNet is a different case anyways. There, the "graph" is only part of the "forward" signature. To run the model in a "graph-parallel" fashion, the user has to define a `GraphPartition` beforehand to setup the `DistributedGraph` properly and then pass this graph structure into the model.

With the current GraphCast design, the graph structure inherently is tied to the model. E.g. the "mesh" itself is often only created within the constructor, i.e. there is no way for the user to define a `GraphPartition` on that graph beforehand anyways.

Reply:
That makes sense and why I said maybe not super important for GraphCast. I guess for now, having booleans to control the partitioning scheme is fine.